compiler/rustc_middle/src/ty/adt.rs

```diff
@@ -117,7 +117,7 @@ impl PartialEq for AdtDef {
     // `AdtDef`s are always interned, and this is part of `TyS` equality.
     #[inline]
```

Comment:
are the `Eq` and `Hash` impls relevant for performance?

because if so, an alternative would be use a different type when dealing with not interned  `AdtDef`s, e.g. by using
```rust
pub struct AdtDef<const INTERNED = true> {
    ...
}
```

Reply:
considering that even all `full` benches are faster now (which i don't really understand), it doesn't seem to matter too much :thinking:

Reply:
I think the change in `full` benchmarks is due to the fact that we encode `AdtDef`s into the crate metadata (as a part of `TyKind`)

Reply:
ah, so it's a tradeoff between the size of the metadata and having to recompute `adt_def` for adts in dependencies when compiling a crate

Reply:
That's right. Surprisingly, this is actually a net disk space win for incr comp in many cases (I checked a few on the benchmarks). The metadata becomes larger, but that's outweighed by the dep graph becoming much smaller.

Reply:
Also, let's move this conversation to https://github.com/rust-lang/rust/pull/91924, which contains just the `AdtDef` changes